internal/framework/provider/functions/encode.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package functions
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"sigs.k8s.io/yaml"
+)
+
+func encodeValue(v attr.Value) (any, error) {
+	switch vv := v.(type) {
+	case basetypes.StringValue:
+		return vv.ValueString(), nil
+	case basetypes.NumberValue:
+		f, _ := vv.ValueBigFloat().Float64()
+		return f, nil
+	case basetypes.Float64Value:
+		return vv.ValueFloat64(), nil
+	case basetypes.Int64Value:
+		return vv.ValueInt64(), nil
```

Comment:
Just an FYI, the only possible branch for all number types is `basetypes.NumberValue` since floats/ints are both represented in Terraform by the underlying `tftypes.Number`.

The logic for how we determine what framework type something should be is here: https://github.com/hashicorp/terraform-plugin-framework/blob/484f311c99cf62e7bbf638ede03e1c7df97151db/types/basetypes/dynamic_type.go#L116

Reply:
Ah ha, interesting thanks! I've removed the cases for Int64 and Float64 and sure enough my tests with floats and ints still pass.